status-handler.js

```diff
@@ -173,6 +182,51 @@ class StatusHandler {
     return true;
   }
 
+  /**
+   * Check if a build is completed for all locales
+   *
+   * @function _isBuildCompleted
+   * @param {Object} data - Message from NSQ
+   * @returns {String} the package name
+   */
+  async _isBuildCompleted(data) {
+    const { version, env, locale } = data;
+    const pkg = this._getPackageName(data);
+
+    const previousEvents = await this.models.StatusEvent.findAll({
+      pkg, version, env
+    });
+
+    // NOTE(jdaeli): this should never occur - TO CONFIRM
+    if (!previousEvents || previousEvents.length === 0) return false;
+
+    const locales = await new Promise((resolve, reject) => {
+      this.wrhs.packages.get({ pkg, version, env }, (err, result) => {
+        if (err) return reject(err);
+        resolve(result.extended.locales || []);
+      });
+    });
+
+    const completedLocales = previousEvents
+      .filter(event => event.message === 'carpenterd-worker build completed')
+      .map(event => event.locale);
+
+    const numOfCompletedLocales = (Array.from(new Set(completedLocales))).length;
+
+    // CASE 1: no previous locales completed, and only 1 package locale to build
+    if (numOfCompletedLocales === 0 && locales.length < 2) return true;
+
+    // CASE 2: all locales completed except the current one
```

Comment:
since we are creating a status event before we run this function, I think we should handle the race condition of whether or not it was inserted.

Reply:
Wait are we?

Reply:
Discussed this offline. Solution is moving the webhook call with a blocking call before the creation of the event in the complete function.

Reply:
Re discussed this again. Applying changes now.